js/show-graph.js

```diff
@@ -184,24 +152,69 @@ diagram.showGraph = function(args) {
+//              .style('stroke', '#'+colorpicker.value)
+//              .style('fill', function(d) { return d.bobbin ? '#'+colorpicker.value : 'none' })
+//        })
+//    }
+
+    var drawPath = function(d) {
+        var source = args.nodes[d.source]
+        var target = args.nodes[d.target]
+        var sX = source.x
+        var sY = source.y
+        var tX = target.x
+        var tY = target.y
+        var dX = (tX - sX)
+        var dY = (tY - sY)
+        var mid = d.left ? ("S" + (sX - dY/3 + dX/3) + "," + (sY + dX/3 + dY/3)) :
+                  d.right ? ("S" + (sX + dY/3 + dX/3) + "," + (sY + dY/3 - dX/3)) :
+                             " "
+        if (d.end && d.end == "white")
+            return "M"+ sX + "," + sY + mid + " " + (tX - dX/4) + "," + (tY - dY/4)
+        if (d.start && d.start == "white")
+            return "M"+ (sX + dX/4) + "," + (sY + dY/4) + mid + " " + tX + "," + tY
+        if (d.mid)
+            return "M"+ sX + "," + sY + " " + (sX + dX/2) + "," + (sY + dY/2) + " " + tX + "," + tY
+        return "M"+ sX + "," + sY + " " + tX + "," + tY
+    }
+    var moveNode = function(d) {
+        return "translate(" + d.x + "," + d.y + ")"
+    }
     // a higher speed for IE as marks only appear when the animation is finished
     var mod = isMobileMac ? 10 : isIE ? 3 : 2
     var step = 0
-    // layout simulation step
-    force.on("tick", function() {
-        if ( ((step++)%mod) != 0) return
-        link.attr("d", diagram.path)
-        node.attr("transform", diagram.transform)
-    })
-    force.on("end", function(){
-      if (isIE) diagram.markLinks(link)
-      if (args.onAnimationEnd) args.onAnimationEnd()
-    })
+    var simTicked = function() {
+                         if ( ((step++)%mod) != 0) return // skip rendering
+                         nodes.attr("transform", moveNode)
+                         links.attr("d", drawPath)
+                     }
+    var simEnded = function(){
+                      if (isIE) diagram.markLinks(links)
+                      if (args.onAnimationEnd) args.onAnimationEnd()
+                    }
+//    var force = d3.layout.force()
+//        .nodes(args.nodes)
+//        .links(args.links)
+//        .size([args.width, args.height])
+//        .charge(-20)
+//        .linkDistance(10)
+//        .linkStrength(10)
+//        .start()
+//        .alpha(0.01)
+// Above calibration with the v3 API resulted in a relative quick compact animation.
+// The power of the v4 API requires more insight.
+// https://github.com/d3/d3/blob/master/CHANGES.md#forces-d3-force
+    var sim = d3.forceSimulation()
+        .force("link", d3.forceLink(links).strength(5).distance(10).iterations(2))
+        .force("charge", d3.forceManyBody().distanceMin(7).distanceMax(25).strength(-20))
+        .force("center", d3.forceCenter(args.width / 4, args.height / 6))
+        .force("collide", d3.forceCollide().radius(8).strength(1))
+        .alpha(2)
```

Comment:
Tried various combination of values
* in some configurations the nodes quickly stop changing their relative position while the diagram keeps inflating
* the current configuration has an irregular result
* the nodes in the top left corner hardly move at all
* never saw a thread diagram that was more or less recognizable